test/memory.jl

```diff
@@ -10,58 +10,81 @@ let
 end
 =#
 
-# pointer-based
-src = 42
+let
+    # pointer-based
+    src = 42
 
-buf1 = Mem.alloc(sizeof(src))
+    buf1 = Mem.alloc(sizeof(src); coherent=true)
 
-Mem.set!(buf1, UInt32(0), sizeof(Int)÷sizeof(UInt32))
+    Mem.set!(buf1, UInt32(57), 1)
+    x = Mem.download(UInt32, buf1)
+    @test x[1] == UInt32(57)
 
-Mem.upload!(buf1, Ref(src), sizeof(src))
+    Mem.upload!(buf1, pointer_from_objref(Ref(src)), sizeof(src))
```

Comment:
I believe that you technically need a `GC.@preserve` for these `pointer_from_objref` calls, because the `Ref` can be GC'd at any time after that call completes (which could be right before the transfer starts).